third_party/move/tools/move-package/src/compilation/build_plan.rs

```diff
@@ -455,7 +455,19 @@
         for dir in std::fs::read_dir(build_root)? {
             let path = dir?.path();
             if !keep_paths.iter().any(|name| path.ends_with(name.as_str())) {
-                std::fs::remove_dir_all(&path)?;
+                if path.is_dir() {
+                    std::fs::remove_dir_all(&path).map_err(|e| {
```

Comment:
I think you can just use `with_context` on the result see the example here:
https://docs.rs/anyhow/latest/anyhow/trait.Context.html